src/main/java/uk/gov/companieshouse/acsp/manage/users/controller/UserAcspMembership.java

```diff
@@ -211,11 +210,68 @@ public ResponseEntity<List<AcspMembership>> getAcspMembershipForUserId(
+    final String requestingUserId = requestingAcspMember.getUserId();
+    final UserRoleEnum requestingUserRole = requestingAcspMember.getUserRole();
+    final String targetUserId = targetAcspMembership.getUserId();
+    final UserRoleEnum targetUserRole = targetAcspMembership.getUserRole();
+
+    if ( requestingUserRole.equals( UserRoleEnum.STANDARD ) ){
+      LOG.error( String.format( "%s: Requesting user %s has a standard role", xRequestId, requestingUserId ) );
+      throw new BadRequestRuntimeException( PLEASE_CHECK_THE_REQUEST_AND_TRY_AGAIN );
+    }
+
+    if ( requestingUserRole.equals( UserRoleEnum.ADMIN ) && targetUserRole.equals( UserRoleEnum.OWNER ) ){
+      LOG.error( String.format( "%s: Requesting user %s has an admin role, and target user %s has an owner role", xRequestId, requestingUserId, targetUserId ) );
+      throw new BadRequestRuntimeException( PLEASE_CHECK_THE_REQUEST_AND_TRY_AGAIN );
+    }
+
+    if ( targetUserRole.equals( UserRoleEnum.OWNER ) && ( targetUserId.equals( requestingUserId ) ) ){
+      LOG.error( String.format( "%s: Requesting user %s is an owner and is attempting to update themselves", xRequestId, requestingUserId ) );
+      throw new BadRequestRuntimeException( PLEASE_CHECK_THE_REQUEST_AND_TRY_AGAIN );
+    }
   }
+
+    @Override
+    public ResponseEntity<Void> updateAcspMembershipForId( final String xRequestId, final String targetAcspMemberId, final RequestBodyPatch requestBodyPatch ) {
+      final var action = requestBodyPatch.getAction();
+      final var targetUserNewRole = requestBodyPatch.getUserRole();
+
+      LOG.infoContext( xRequestId, String.format( "Attempting to perform %s operation on AcspMember %s", action, targetAcspMemberId ), null );
+
+      final var targetAcspMembership =
+      acspMembersService.fetchAcspMembersDao( targetAcspMemberId )
+                        .orElseThrow( () -> {
+                            LOG.error( String.format( "%s: AcspMember %s does not exist", xRequestId, targetAcspMemberId ) );
+                            return new NotFoundRuntimeException( StaticPropertyUtil.APPLICATION_NAMESPACE, PLEASE_CHECK_THE_REQUEST_AND_TRY_AGAIN );
+                        } );
+
+      final var requestingUserId = UserContext.getLoggedUser().getUserId();
+      final var requestingAcspMember =
+      acspMembersService.fetchActiveAcspMemberByUserIdAndAcspNumber( requestingUserId, targetAcspMembership.getAcspNumber() )
+                        .orElseThrow( () -> {
+                            LOG.error( String.format( "%s: Active AcspMember where user_id=%s and acsp_number=%s does not exist", xRequestId, requestingUserId, targetAcspMembership.getAcspNumber() ) );
+                            return new BadRequestRuntimeException( PLEASE_CHECK_THE_REQUEST_AND_TRY_AGAIN );
+                        } );
+
+      toBadRequestWhenActionIsNotPermitted( xRequestId, requestingAcspMember, targetAcspMembership );
```

Comment:
Small one, but the `to` initially made me think it was converting something `from` something `to` something, I think `throw` might be better to indicate the methods side-effect.

```suggestion
      throwBadRequestWhenActionIsNotPermitted( xRequestId, requestingAcspMember, targetAcspMembership );
```

Reply:
Good point. I've made the change